### ISSUE TITLE
Fix `fill_cache` etc.

### ISSUE THREAD

**Issue by ranocha** [ranocha/CLBlast.jl#13]
Currently, I get
```
julia> using OpenCL, CLBlast

julia> device, ctx, queue = cl.create_compute_context()
(OpenCL.Device(GeForce GTX 1070 Ti on NVIDIA CUDA @0x000056360f4bbda0), OpenCL.Context(@0x000056360f30fe70 on GeForce GTX 1070 Ti), OpenCL.CmdQueue(@0x00005636105c7fa0))

julia> CLBlast.fill_cache(device)
0

julia> pointer(device)
Ptr{Void} @0x000056360f4bbda0

julia> x = rand(Float32, 10); x_cl = cl.CLArray(queue, x)
CLArray{Float32,1}(Buffer{Float32}(@0x00005636144505a0),(10,))

julia> CLBlast.nrm2(length(x_cl), x_cl, 1)

signal (11): Segmentation fault
while loading no file, in expression starting on line 0
unknown function (ip: 0xffffffffffffffff)
_ZN7clblast7RoutineC1ERNS_5QueueEPP9_cl_eventRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorISB_SaISB_EENS_9PrecisionERKSE_INS_8database13DatabaseEntryESaISL_EESt16initializer_listIPKcE at /home/hendrik/.julia/v0.6/CLBlast/deps/CLBlast-1.4.0-Linux-x64/lib/libclblast.so (unknown line)
unknown function (ip: 0x7665447465476c62)
Allocations: 2280045 (Pool: 2278735; Big: 1310); GC: 2
Segmentation fault (core dumped)
```

**Comment by ranocha** [comment]
Another example:
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.3-pre.0 (2017-12-18 07:11 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 93168a6826 (192 days old release-0.6)
|__/                   |  x86_64-linux-gnu

julia> using OpenCL, CLBlast

julia> device, ctx, queue = cl.create_compute_context()
(OpenCL.Device(pthread-Intel(R) Core(TM) i5-4590S CPU @ 3.00GHz on Portable Computing Language @0x000056185355d750), OpenCL.Context(@0x0000561852748b80 on pthread-Intel(R) Core(TM) i5-4590S CPU @ 3.00GHz), OpenCL.CmdQueue(@0x000056185313fb30))

julia> CLBlast.fill_cache(device)
0

julia> CLBlast.fill_cache(device)
ERROR: ReadOnlyMemoryError()

julia> CLBlast.fill_cache(device)

signal (11): Speicherzugriffsfehler
while loading no file, in expression starting on line 0
unknown function (ip: 0xfffffffff)
Allocations: 2273670 (Pool: 2272404; Big: 1266); GC: 2
Speicherzugriffsfehler
```
Whether there is a `ReadOnlyMemoryError` or a segfault seems to be random...

**Comment by ranocha** [comment]
This seems to be fixed by updating to CLBlast v1.4.1.
